### PR TITLE
feat: unify account controllers with service layer

### PR DESCRIPTION
**Pull request by polillomm**
Some comments made in this **PR** will be addressed even after the _merge_. Therefore, I am leaving them open and unresolved, using them as a guide.

### CODE REVIEW

File: src/domain/valueObject/accountId.go [hunk 1]
```diff
@@ -10,14 +10,15 @@ import (
...
 	if err != nil {
-		return 0, errors.New("InvalidAccountId")
+		return 0, errors.New("AccountIdMustBeInt")
 	}
 
-	return AccountId(accId), nil
+	return AccountId(accountId), nil
 }
 
+// TODO: remove this constructor when no longer used.
```
Comment (polillomm) [inline src/domain/valueObject/accountId.go:21]: This _TODO_ comment will be removed once the other components that use this panic constructor are refactored and no longer rely on it.

File: src/domain/valueObject/password.go [hunk 2]
```diff
@@ -28,6 +26,7 @@ func NewPassword(value interface{}) (password Password, err error) {
 	return Password(stringValue), nil
 }
 
+// TODO: remove this constructor when no longer used.
```
Comment (polillomm) [inline src/domain/valueObject/password.go:29]: Same as [this comment](https://github.com/speedianet/os/pull/142/files#r1683023375).

File: src/domain/valueObject/username.go [hunk 3]
```diff
@@ -3,33 +3,37 @@ package valueObject
...
 	}
-	return user, nil
+	return Username(stringValue), nil
 }
 
-func NewUsernamePanic(value string) Username {
-	user := Username(value)
-	if !user.isValid() {
-		panic("InvalidUsername")
+// TODO: remove this constructor when no longer used.
```
Comment (polillomm) [inline src/domain/valueObject/username.go:28]: Same as [this comment](https://github.com/speedianet/os/pull/142/files#r1683023375).

File: src/domain/valueObject/accountId.go (hunk 1 above)
Comment (ntorga) [reply src/domain/valueObject/accountId.go:21]: Right, you don't need to mention that on the code and push it to the main. You can just add to your personal to-do file or on that task to replace panics. Please, remove the todo comment from the code.

File: src/domain/valueObject/accountId.go [hunk 4]
```diff
@@ -26,10 +27,10 @@ func NewAccountIdPanic(value interface{}) AccountId {
 	return accId
 }
 
-func (id AccountId) Get() uint64 {
-	return uint64(id)
+func (vo AccountId) Get() uint64 {
```
Comment (ntorga) [inline src/domain/valueObject/accountId.go:30]: Also add on the task to replace Get() with Read() as we're already doing on newer code.

File: src/domain/valueObject/password.go (hunk 2 above)
Comment (ntorga) [reply src/domain/valueObject/password.go:29]: And same for what I mentioned above, hehe. Won't be repeating, so please remove this TODOs lines.

File: src/presentation/api/controller/account.go [hunk 5]
```diff
@@ -0,0 +1,92 @@
+package apiController
+
+import (
+	"github.com/labstack/echo/v4"
+	apiHelper "github.com/speedianet/os/src/presentation/api/helper"
+	"github.com/speedianet/os/src/presentation/service"
+)
+
+type AccountController struct {
+	accountService service.AccountService
```
Comment (ntorga) [inline src/presentation/api/controller/account.go:10]: This should be a pointer.

File: src/presentation/cli/controller/account.go [hunk 6]
```diff
@@ -0,0 +1,118 @@
+package cliController
+
+import (
+	cliHelper "github.com/speedianet/os/src/presentation/cli/helper"
+	"github.com/speedianet/os/src/presentation/service"
+	"github.com/spf13/cobra"
+)
+
+type AccountController struct {
+	accountService service.AccountService
```
Comment (ntorga) [inline src/presentation/cli/controller/account.go:10]: Also should be a pointer.

File: src/presentation/service/account.go [hunk 7]
```diff
@@ -0,0 +1,157 @@
...
+	filesQueryRepo := filesInfra.FilesQueryRepo{}
+	filesCmdRepo := filesInfra.FilesCmdRepo{}
+
+	err = useCase.CreateAccount(
+		accQueryRepo,
+		accCmdRepo,
+		filesQueryRepo,
+		filesCmdRepo,
+		dto,
+	)
```
Comment (ntorga) [inline src/presentation/service/account.go:59]: We are not breaking lines like this anymore.

File: src/presentation/service/account.go [hunk 8]
```diff
@@ -0,0 +1,157 @@
...
+		shouldUpdateApiKey := input["shouldUpdateApiKey"].(bool)
+		shouldUpdateApiKeyPtr = &shouldUpdateApiKey
+	}
+
+	dto := dto.NewUpdateAccount(
+		accountIdPtr,
+		usernamePtr,
+		passwordPtr,
+		shouldUpdateApiKeyPtr,
+	)
```
Comment (ntorga) [inline src/presentation/service/account.go:106]: Please remove the unnecessary break lines every time you see them.

**Review by ntorga — CHANGES_REQUESTED** [review]

File: src/domain/valueObject/accountId.go (hunk 1 above)
Comment (polillomm) [reply src/domain/valueObject/accountId.go:21]: Done.

File: src/domain/valueObject/password.go (hunk 2 above)
Comment (polillomm) [reply src/domain/valueObject/password.go:29]: Done.

File: src/domain/valueObject/username.go (hunk 3 above)
Comment (polillomm) [reply src/domain/valueObject/username.go:28]: Done.

File: src/domain/valueObject/accountId.go (hunk 4 above)
Comment (polillomm) [reply src/domain/valueObject/accountId.go:30]: Done.

File: src/presentation/api/controller/account.go (hunk 5 above)
Comment (polillomm) [reply src/presentation/api/controller/account.go:10]: Done.

File: src/presentation/cli/controller/account.go (hunk 6 above)
Comment (polillomm) [reply src/presentation/cli/controller/account.go:10]: Done.

File: src/presentation/service/account.go (hunk 7 above)
Comment (polillomm) [reply src/presentation/service/account.go:59]: Done.

File: src/presentation/service/account.go (hunk 8 above)
Comment (polillomm) [reply src/presentation/service/account.go:106]: Done.

**Pull request merged**